File: 1910-remove-all-occurrences-of-a-substring/1910-remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    bool check(stack<int> st,string& part){
        if(st.size()<part.size()) return false;
        string temp;
        while(temp.size()<part.size()) {
            temp.push_back(st.top());
            st.pop();
        }
        reverse(temp.begin(),temp.end());
        return temp==part;
    }
    string removeOccurrences(string s, string part) {
        stack<int> st;
        for(int i=0; i<s.size(); ++i){
            st.push(s[i]);
            while(st.size()>0 && st.top()==part[part.size()-1] && check(st,part)) {
                int temp = part.size();
                while(temp--) st.pop();
            }
        }
        string ans;
        while(!st.empty()) {
            ans.push_back(st.top());
            st.pop();
        }
        reverse(ans.begin(),ans.end());
        return ans;
    }
};

File: 1910-remove-all-occurrences-of-a-substring/1910-remove-all-occurrences-of-a-substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        Stack<Character> st = new Stack<>();
        
        for(int i=0;i<s.length();i++){
            char ch = s.charAt(i);
            st.push(ch);
            
            if(st.size() >= part.length() && st.size() != 0 && st.peek() == part.charAt(part.length()-1)){
                int idx = i-1;
                boolean flag = true;
                Stack<Character> temp = new Stack<>();
                for(int j=part.length()-1;j>=0;j--){
                    if(part.charAt(j) == st.peek()){
                        temp.push(st.pop());
                    } else {
                        flag = false;
                        break;
                    }
                }
                
                if(!flag){
                    while(temp.size() != 0){
                        st.push(temp.pop());
                    }
                }
                
            }
            
            
        }
        
        StringBuilder sb = new StringBuilder();
        while(st.size() != 0){
            sb.append(st.pop());
        }
        
        sb.reverse();
        
        return sb.toString();
        
    }
}